src/Nethermind/Nethermind.Facade/Filters/LogOperators.cs

```diff
@@ -0,0 +1,93 @@
+using System.Collections.Generic;
+using Nethermind.Core.Collections;
+
+namespace Nethermind.Blockchain.Filters.Topics
+{
+
+    public class LogOperators<Element>
+    {
+
+        public static IEnumerable<Element> Intersect(IEnumerator<Element>[] enumerators)
+        {
+
+            DictionarySortedSet<Element, IEnumerator<Element>> transactions = new();
+
+            for (int i = 0; i < enumerators.Length; i++)
+            {
+                IEnumerator<Element> enumerator = enumerators[i];
+                if (enumerator.MoveNext())
+                {
+                    transactions.Add(enumerator.Current!, enumerator);
+                }
+            }
+
+
+            while (transactions.Count == enumerators.Length)
+            {
+                (Element blockNumber, IEnumerator<Element> enumerator) = transactions.Min;
+                (Element blockNumber2, IEnumerator<Element> enumerator2) = transactions.Max;
+
+                bool isIntersection = EqualityComparer<Element>.Default.Equals(blockNumber, blockNumber2);
+
+                transactions.Remove(blockNumber);
+
+                if (enumerator.MoveNext())
+                {
+                    transactions.Add(enumerator.Current!, enumerator);
+                }
```

Comment:
Don't you want to advance ALL values that are smaller than Max?

Reply:
I believe the result would be the same since there wont be any intersection until the values are greater than the Max but yea it will prevent the extra need to getting the .Max everytime